src/Assigning/Assignment.hs

```diff
@@ -369,7 +368,7 @@ instance (Enum grammar, Eq1 ast, Ix grammar, Show grammar, Show1 ast) => Parsing
   eof = tracing End `Then` pure
 
   notFollowedBy :: (HasCallStack, Show a) => Assignment ast grammar a -> Assignment ast grammar ()
-  notFollowedBy a = a *> unexpected (show a) <|> pure ()
+  notFollowedBy a = (a >>= unexpected . show) <|> pure ()
```

Comment:
This enables us to avoid defining `Show1` machinery for `Freer`; furthermore, the method was always designed to show the result of its argument, and not the argument itself, so this has been wrong for ~ever.